AraEvent/AraEventCalibrator.cxx

```diff
@@ -1458,11 +1491,159 @@ Bool_t AraEventCalibrator::TrimFirstBlock(UsefulAtriStationEvent *theEvent, std:
                 voltMapIt->second.erase(voltMapIt->second.begin(),voltMapIt->second.begin()+first_block_len); ///< delete the times in the first block
                 timeMapIt->second.erase(timeMapIt->second.begin(),timeMapIt->second.begin()+first_block_len); ///< delete the ADC (or volt) in the first block
                 sampleList->at(chanId).erase(sampleList->at(chanId).begin(),sampleList->at(chanId).begin()+first_block_len); ///< delete the samples in the first block
+                blockList->at(chanId).erase(blockList->at(chanId).begin(),blockList->at(chanId).begin()+first_block_len); ///< delete the samples in the first block
             }
         }
     } return !hasTooFewBlocks;
 }
 
+//! Correct block offsets
```

Comment:
Can you add a little more documentation, in words, describing what this function is trying to do at a high level? Like "We are going to loop over every block, and FFT it. Then we are going to remove the low frequency modes." Or something like that? I think it's complicated enough that we want some high level docs.

Reply:
In particular some words about why/how this is being cast as a matrix problem. I don't think that will be obvious to a future user.

Reply:
Okay how's this:

```
This function models each block offset as a rectangle function in the time-domain, solving for the amplitude of the function in the
frequency-domain. It does this by transforming a sum of rectangle functions into the frequency-domain and setting their amplitudes to
minimize the squared error between this model and the low frequency spectrum of the waveform. Only the low frequency spectrum is used
in order to minimize the impact on real signals. Each channel's block offsets are fit independently.

Since the model for block offsets is linear in their amplitudes, the squared error is convex in the amplitudes and we can solve for the optimal 
amplitudes exactly (by taking the derivative of the squared error and setting it to zero). This results in a system of linear equations which 
can be solved via a matrix equation: M.A = b, where M is a matrix of constant coefficients, A is a vector of block offset amplitudes, and b is
a vector of constants. 
```

Reply:
Yeah, I think this is nice! I'm not sure it's obvious to me why M and b are constants? But maybe that's a detail. Perhaps you can give a talk on the analysis call about this? I think it's worth having some real docs. And then they can be linked.

Reply:
Ah, M and b are just the constants in the system of linear equations that needs to be solved. Maybe there's a better way to phrase it.

And sure, I can try to pull that together! Though probably it won't happen until after the New Year at this point. Or I can present it on another call.

Reply:
Oh, I see. Maybe just "where M and b are the constants of the system of linear equations which need to be solved." I like exactly what you said.

And sure, after new year's sounds good!